agent/engine/docker_task_engine_test.go

```diff
@@ -3128,3 +3128,252 @@ func TestStartFirelensContainerRetryForContainerIP(t *testing.T) {
+			resetExecMetadata: false,
+		},
+		{
+			topResponse:       nil,
+			topError:          errors.New(dockerapi.TopProcessNotFoundErrorName),
+			resetExecMetadata: true,
+		},
+		{
+			topResponse: &dockercontainer.ContainerTopOKBody{
+				Processes: [][]string{},
+			},
+			topError:          nil,
+			resetExecMetadata: true,
+		},
+		{
+			topResponse:       nil,
+			topError:          errors.New("no container error"),
+			resetExecMetadata: false,
+		},
+	}
+	for _, tc := range testCases {
+		testTask := &apitask.Task{
+			Arn: "arn:aws:ecs:region:account-id:task/test-task-arn",
+			Containers: []*apicontainer.Container{
+				{
+					Name:      "test-container",
+					RuntimeID: "runtime-ID",
+					ExecCommandAgentMetadata: &apicontainer.ExecCommandAgentMetadata{
+						PID: execAgentPID,
+					},
+				},
+			},
+			ExecCommandAgentEnabled: true,
+		}
+		taskEngine.(*DockerTaskEngine).state.AddTask(testTask)
+		client.EXPECT().TopContainer(gomock.Any(), testTask.Containers[0].RuntimeID, gomock.Any(), execAgentPID).Return(tc.topResponse, tc.topError)
+		taskEngine.(*DockerTaskEngine).monitorExecAgentRunning(ctx, testTask.Arn, testTask.Containers[0])
+		if tc.resetExecMetadata {
+			assert.Equal(t, "", testTask.Containers[0].GetExecCommandAgentMetadata().PID)
+		} else {
+			assert.Equal(t, execAgentPID, testTask.Containers[0].GetExecCommandAgentMetadata().PID)
+		}
+	}
+}
+
+func TestMonitorExecAgentProcesses(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, client, _, taskEngine, _, _, _ := mocks(t, ctx, &defaultConfig)
+	defer ctrl.Finish()
+	execAgentPID := "1234"
+	resp := &dockercontainer.ContainerTopOKBody{
+		Processes: [][]string{{"root", execAgentPID}},
+	}
+	testTask := &apitask.Task{
+		Arn: "arn:aws:ecs:region:account-id:task/test-task-arn",
+		Containers: []*apicontainer.Container{
+			{
+				Name:      "test-container",
+				RuntimeID: "runtime-ID",
+				ExecCommandAgentMetadata: &apicontainer.ExecCommandAgentMetadata{
+					PID: execAgentPID,
+				},
+			},
+		},
+		ExecCommandAgentEnabled: true,
+	}
+	taskEngine.(*DockerTaskEngine).state.AddTask(testTask)
+	taskEngine.(*DockerTaskEngine).managedTasks[testTask.Arn] = &managedTask{Task: testTask}
+	client.EXPECT().TopContainer(gomock.Any(), testTask.Containers[0].RuntimeID, gomock.Any(), execAgentPID).Return(resp, nil)
+	taskEngine.(*DockerTaskEngine).monitorExecAgentProcesses(ctx)
+	time.Sleep(5 * time.Millisecond)
+	assert.Equal(t, execAgentPID, testTask.Containers[0].GetExecCommandAgentMetadata().PID)
+}
+
+func TestMonitorExecAgentProcessExecDisabled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, _, _, taskEngine, _, _, _ := mocks(t, ctx, &defaultConfig)
+	defer ctrl.Finish()
+	testTask := &apitask.Task{
+		Arn: "arn:aws:ecs:region:account-id:task/test-task-arn",
+		Containers: []*apicontainer.Container{
+			{
+				Name:      "test-container",
+				RuntimeID: "runtime-ID",
+			},
+		},
```

Comment:
nit: I know the zero value for ExecCommandAgentEnabled is false, but just for the sake of explicitness, could we add `ExecCommandAgentEnabled: false,` here?